src/Lykke.SettingsReader/SettingsProcessor/SettingsProcessor.cs

```diff
@@ -119,31 +118,63 @@ private static string ConcatPath(string path, string propertyName)
             return $"{path}.{propertyName}".Trim('.');
         }
 
-        private static void ProcessChecks<T>(T model)
+        private static int CheckPropertiesCount<T>(T model)
         {
-            var result = new StringBuilder();
-            var properties = (from p in model.GetType().GetTypeInfo().GetProperties(BindingFlags.Public | BindingFlags.Instance)
-                where p.CanWrite && p.CanRead && Attribute.IsDefined(p, typeof(BaseCheckAttribute)) select p).ToList();
+            int count = 0;
 
-            if (!properties.Any())
-                return;
+            if (model == null)
+                return count;
 
-            Console.WriteLine("Checking services");
-
-            foreach (var property in properties)
+            Type objType = model.GetType();
+            PropertyInfo[] properties = objType.GetProperties(BindingFlags.Public | BindingFlags.Instance);
+            
+            foreach (PropertyInfo property in properties)
             {
                 object value = property.GetValue(model);
-
                 var checkAttribute = (BaseCheckAttribute)property.GetCustomAttribute(typeof(BaseCheckAttribute));
 
-                var checker = checkAttribute.GetChecker();
+                if (checkAttribute != null)
+                {
+                    count++;
+                }
+                else if (property.PropertyType.IsClass && !property.PropertyType.IsValueType &&
+                         !property.PropertyType.IsPrimitive && property.PropertyType.FullName != typeof(string).FullName
```

Comment:
Will be better to replace name comparison with the type comparison (property.PropertyType != typeof(string). The same notice for the object.